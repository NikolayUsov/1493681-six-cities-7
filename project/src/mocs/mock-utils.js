const PHRASE = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra. Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante. Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui. Sed sed nisi sed augue convallis suscipit in sed felis. Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.';

export const getRandomInteger = (a = 0, b = 1) => {
  const lower = Math.ceil(Math.min(a, b));
  const upper = Math.floor(Math.max(a, b));

  return Math.floor(lower + Math.random() * (upper - lower + 1));
};

export const createId = (start = 0) => {
  let id = start;
  return () => {
    id += 1;
    return id;
  };
};

export const getRandomFloat = (min, max, point = 1) => (Math.random() * (max - min) + min)
  .toFixed(point);

export const shuffleArray = (array) => {
  const newArr = array.slice();
  for (let i = newArr.length - 1; i > 0; i -= 1) {
    const j = Math.floor(Math.random() * (i + 1));
    [newArr[i], newArr[j]] = [newArr[j], newArr[i]];
  }

  return newArr;
};

export const getRandomElement = (arr) => arr[getRandomInteger(0, arr.length - 1)];

export const getRandomText = () => shuffleArray(PHRASE.split('.')).slice(0, getRandomInteger(1, 5)).join('.');
