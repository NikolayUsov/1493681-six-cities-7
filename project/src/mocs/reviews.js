import dayjs from 'dayjs';
import {
  getRandomText,
  getRandomInteger,
  createId,
  getRandomFloat,
  getRandomElement,
} from './mock-utils';

const AVATAR_URL = 'https://i.pravatar.cc/';
const commentId = createId();
const NAMES = ['Fedor', 'Angelina', 'Gena', 'Slava', 'Pedro', 'Roma'];

const review = () => ({
  comment: getRandomText(),
  date: dayjs().subtract(getRandomInteger(1, 25), 'year'),
  id: commentId(),
  rating: getRandomFloat(0, 5),
  user: {
    avatarUrl: `${AVATAR_URL}${getRandomInteger(1, 50)}`,
    isPro: Boolean(getRandomInteger(0, 1)),
    name: getRandomElement(NAMES),
    id: getRandomInteger(100, 900),
  },
});

// eslint-disable-next-line import/prefer-default-export
export const createRandomReviews = () => new Array(getRandomInteger(1, 10)).fill().map(review);
