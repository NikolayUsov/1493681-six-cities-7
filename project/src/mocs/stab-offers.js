const offersStab = [
  {
    'bedrooms':3,
    'city':{
      'location':{
        'lat':52.95029,
        'lng':39.6850183,
        'zoom':10,
      },
      'name':'Brussels',
    },
    'description':' Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Cras aliquet varius magna, non porta ligula feugiat eget. Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.. Nunc fermentum tortor ac porta dapibus',
    'goods':[
      'Dishwasher',
      'Kitchen',
      'Heating',
      'Coffee machine',
      'Cable TV',
      'Washing machine',
    ],
    'host':{
      'avatarUrl':'https://i.pravatar.cc/2',
      'id':1,
      'isPro':false,
      'name':'Fedor',
    },
    'id':1,
    'images':[
      'img/apartment-01.jpg',
    ],
    'isFavorite':true,
    'isPremium':true,
    'location':{
      'lat':52.9392527,
      'lng':4.0813024,
      'zoom':5,
    },
    'maxAdults':1,
    'previewImage':'img/apartment-01.jpg',
    'price':139,
    'rating':1.4,
    'title':'Lorem ipsum dolo',
    'type':'apartment',
  },
  {
    'bedrooms':3,
    'city':{
      'location':{
        'lat':52.9827976,
        'lng':25.1066787,
        'zoom':10,
      },
      'name':'Paris',
    },
    'description':' Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante',
    'goods':[
      'Dishwasher',
      'Coffee machine',
      'Cable TV',
      'Kitchen',
      'Heating',
      'Washing machine',
    ],
    'host':{
      'avatarUrl':'https://i.pravatar.cc/45',
      'id':2,
      'isPro':true,
      'name':'Gena',
    },
    'id':2,
    'images':[
      'img/apartment-03.jpg',
      'img/apartment-02.jpg',
      'img/apartment-01.jpg',
      'img/apartment-02.jpg',
      'img/apartment-03.jpg',
      'img/apartment-01.jpg',
    ],
    'isFavorite':false,
    'isPremium':true,
    'location':{
      'lat':52.5799583,
      'lng':4.4108526,
      'zoom':7,
    },
    'maxAdults':5,
    'previewImage':'img/apartment-03.jpg',
    'price':132,
    'rating':1.1,
    'title':' Fusce tristique',
    'type':'room',
  },
  {
    'bedrooms':3,
    'city':{
      'location':{
        'lat':52.8408641,
        'lng':31.6061469,
        'zoom':10,
      },
      'name':'Hamburg',
    },
    'description':' Aliquam erat volutpat. Cras aliquet varius magna, non porta ligula feugiat eget. Aliquam id orci ut lectus varius viverra',
    'goods':[
      'Washing machine',
      'Cable TV',
      'Heating',
      'Kitchen',
      'Coffee machine',
      'Dishwasher',
    ],
    'host':{
      'avatarUrl':'https://i.pravatar.cc/37',
      'id':3,
      'isPro':false,
      'name':'Angelina',
    },
    'id':3,
    'images':[
      'img/apartment-03.jpg',
      'img/apartment-02.jpg',
      'img/apartment-02.jpg',
      'img/apartment-03.jpg',
      'img/apartment-02.jpg',
      'img/apartment-01.jpg',
    ],
    'isFavorite':false,
    'isPremium':false,
    'location':{
      'lat':52.6601074,
      'lng':4.9871972,
      'zoom':1,
    },
    'maxAdults':4,
    'previewImage':'img/apartment-03.jpg',
    'price':70,
    'rating':1.9,
    'title':' Cras aliquet va',
    'type':'apartment',
  },
  {
    'bedrooms':4,
    'city':{
      'location':{
        'lat':52.819066,
        'lng':24.4150771,
        'zoom':10,
      },
      'name':'Paris',
    },
    'description':'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    'goods':[
      'Dishwasher',
      'Coffee machine',
      'Kitchen',
      'Washing machine',
      'Cable TV',
      'Heating',
    ],
    'host':{
      'avatarUrl':'https://i.pravatar.cc/46',
      'id':4,
      'isPro':true,
      'name':'Fedor',
    },
    'id':4,
    'images':[
      'img/apartment-03.jpg',
      'img/apartment-03.jpg',
      'img/apartment-02.jpg',
      'img/apartment-02.jpg',
      'img/apartment-02.jpg',
    ],
    'isFavorite':true,
    'isPremium':true,
    'location':{
      'lat':52.3894331,
      'lng':4.5960513,
      'zoom':5,
    },
    'maxAdults':1,
    'previewImage':'img/apartment-03.jpg',
    'price':93,
    'rating':0.7,
    'title':' Cras aliquet va',
    'type':'room',
  },
  {
    'bedrooms':3,
    'city':{
      'location':{
        'lat':52.3529802,
        'lng':20.634975,
        'zoom':10,
      },
      'name':'Cologne',
    },
    'description':' In rutrum ac purus sit amet tempus. Sed sed nisi sed augue convallis suscipit in sed felis. Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante',
    'goods':[
      'Kitchen',
      'Washing machine',
      'Coffee machine',
      'Dishwasher',
      'Cable TV',
      'Heating',
    ],
    'host':{
      'avatarUrl':'https://i.pravatar.cc/3',
      'id':5,
      'isPro':false,
      'name':'Gena',
    },
    'id':5,
    'images':[
      'img/apartment-03.jpg',
      'img/apartment-01.jpg',
      'img/apartment-01.jpg',
    ],
    'isFavorite':true,
    'isPremium':true,
    'location':{
      'lat':52.6285335,
      'lng':4.9916076,
      'zoom':4,
    },
    'maxAdults':2,
    'previewImage':'img/apartment-03.jpg',
    'price':92,
    'rating':1.3,
    'title':' Cras aliquet va',
    'type':'room',
  },
  {
    'bedrooms':2,
    'city':{
      'location':{
        'lat':52.6026323,
        'lng':46.1740771,
        'zoom':10,
      },
      'name':'Hamburg',
    },
    'description':'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam erat volutpat. In rutrum ac purus sit amet tempus.. Nunc fermentum tortor ac porta dapibus',
    'goods':[
      'Washing machine',
      'Cable TV',
      'Heating',
      'Kitchen',
      'Coffee machine',
      'Dishwasher',
    ],
    'host':{
      'avatarUrl':'https://i.pravatar.cc/26',
      'id':6,
      'isPro':true,
      'name':'Angelina',
    },
    'id':6,
    'images':[
      'img/apartment-01.jpg',
      'img/apartment-03.jpg',
    ],
    'isFavorite':false,
    'isPremium':true,
    'location':{
      'lat':52.6109414,
      'lng':4.5352841,
      'zoom':5,
    },
    'maxAdults':6,
    'previewImage':'img/apartment-01.jpg',
    'price':114,
    'rating':3.8,
    'title':'Lorem ipsum dolo',
    'type':'hotel',
  },
];

export { offersStab };
