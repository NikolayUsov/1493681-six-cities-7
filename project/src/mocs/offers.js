
import { getRandomFloat, getRandomInteger, getRandomText, createId} from './mock-utils.js';
const AVATAR_URL = 'https://i.pravatar.cc/';
const CITYES= [
  'Paris',
  'Cologne',
  'Brussels',
  'Amsterdam',
  'Hamburg',
  'Dusseldorf',
];

const NAMES = ['Fedor', 'Angelina', 'Gena', 'Slava'];
const GOODS = ['Heating', 'Kitchen', 'Cable TV', 'Washing machine', 'Coffee machine', 'Dishwasher'];

const Types = {
  apartment: 'Apartment',
  room: 'Private Room',
  house: 'House',
  hotel: 'Hotel',
};

const getHostId = createId(1);
const getOffersId = createId(1);

const generateOfferItemMock = () => {
  const photos = new Array(getRandomInteger(0,6))
    .fill('')
    .map((elem) => `img/${getRandomInteger(1, 3)}.png`);

  return {
    bedrooms: getRandomInteger(1, 4),
    city:{
      location: {
        latitude: getRandomFloat(52, 53),
        longitude: getRandomFloat(4, 5),
        zoom: getRandomInteger(1, 10),
      },
      name: CITYES.randomElement(),
    },
    description: getRandomText(),
    goods: GOODS.shuffle(),
    host: {
      avatarUrl: `${AVATAR_URL}${getRandomInteger(1, 50)}`,
      id: getHostId(),
      isPro: Boolean(getRandomInteger(0, 1)),
      name: NAMES.randomElement(),
    },
    id: getOffersId(),
    images: photos,
    isFavorite:Boolean(getRandomInteger(0, 1)),
    isPremium: Boolean(getRandomInteger(0, 1)),
    location: {
      latitude: getRandomFloat(52, 53),
      longitude: getRandomFloat(4, 5),
      zoom: getRandomInteger(1, 10),
    },
    maxAdults: getRandomInteger(1, 6),
    previewImage: photos.randomElement(),
    price: getRandomInteger(69, 140),
    rating: Number(getRandomFloat(0, 5)),
    title: getRandomText().slice(0, 16),
    type: Object.keys(Types).randomElement(),
  };
};

const offers = new Array(10).fill('').map(generateOfferItemMock);

export { offers };
