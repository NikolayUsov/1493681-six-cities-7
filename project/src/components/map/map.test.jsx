import React from 'react';
import { render } from '@testing-library/react';
import Map from './map';
import  { offersStab } from './../../mocs/stab-offers';
import { MemoryRouter } from 'react-router';

describe('Test Map component', () => {
  const fakeProps = {
    city: {
      lat: 52.9392527,
      lng: 4.0813024,
      zoom:5,
    },
    offers: offersStab,
    activeOffer: offersStab[0],
  };

  it('Should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Map {...fakeProps} />
      </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
