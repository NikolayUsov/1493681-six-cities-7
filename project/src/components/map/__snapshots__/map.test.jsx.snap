// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Map component Should render correctly 1`] = `
<DocumentFragment>
  <section
    class="map cities__map leaflet-container leaflet-grab leaflet-touch-drag"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://c.basemaps.cartocdn.com/rastertiles/voyager/5/16/10.png"
              style="width: 256px; height: 256px; left: -93px; top: -111px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin-active.svg"
          style="margin-left: -15px; margin-top: -40px; width: 30px; height: 40px; left: 0px; top: 0px; z-index: 0;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -15px; margin-top: -40px; width: 30px; height: 40px; left: 7px; top: 13px; z-index: 13;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -15px; margin-top: -40px; width: 30px; height: 40px; left: 20px; top: 10px; z-index: 10;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -15px; margin-top: -40px; width: 30px; height: 40px; left: 12px; top: 20px; z-index: 20;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -15px; margin-top: -40px; width: 30px; height: 40px; left: 21px; top: 12px; z-index: 12;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -15px; margin-top: -40px; width: 30px; height: 40px; left: 10px; top: 12px; z-index: 12;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            +
          </a>
          <a
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            −
          </a>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
